CRM/Utils/Rule.php

```diff
@@ -487,14 +487,23 @@ public static function commaSeparatedIntegers($value) {
    * @return bool
    */
   public static function numeric($value) {
-    // lets use a php gatekeeper to ensure this is numeric
     if (!is_numeric($value)) {
       return FALSE;
     }
 
+    // Don't allow all the variations that PHP does (e.g. scientific notation)
     return (bool) preg_match('/(^-?\d\d*\.\d*$)|(^-?\d\d*$)|(^-?\.\d\d*$)/', $value);
   }
 
+  /**
+   * @param mixed $value
+   *
+   * @return bool
+   */
+  public static function localeNumeric($value) {
```

Comment:
So I kept thinking 'why don't we just make 'numeric' handle formatted numbers like `money` does. I think we could even literally call the money rule (I don't think it enforces a number of decimal places). Then I realised it is not called because we aren't registering it & it is calling a QuickForm default rule.

I'm inclined to think we should register it & call the `money` rule internally & that having it apply like that in the places it doesn't already would be a good thing.... - especially since @mlutfy says there are lots of cases of this